docs/rst/tutorials/core/deployment/keys/topic_keys.rst

```diff
@@ -0,0 +1,234 @@
+a user can also create custom messages using the `IDL (Interface Definition Language) <https://www.omg.org/spec/IDL/4.2/About-IDL>`_
+format following a standardized way of defining the message structure.
+Next example depicts how to define a custom message using the IDL format in ROS 2:
+
+.. code-block:: bash
+
+    # ExampleMsg.idl
+    module package_name {
+      module msg {
+        struct ExampleMsg {
+          long field;
+          short another_field;
+          boolean yet_another_field;
+        };
+      };
+    };
+
+As it can be seen, it is required to name the upper module with the same name as the package name containing it and
+with a nested module called *msg*. Within the *msg* module, user can define the message structure as desired using the IDL syntax
+and name it with the same name as the file.
+
+One of the advantages of defining messages in Interface Definition Language (IDL) is the ability to use annotations.
+Annotations are metadata that provide additional information about IDL constructs such as modules,
+interfaces, operations, attributes, and data types and are used to convey details that are not part of the core
+IDL syntax but are relevant for code generation, documentation, or other purposes.
+Annotations in IDL typically follow the ``@`` symbol and can be applied to various IDL constructs.
+
+Creating Keyed Messages
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Keyed topics typically consist of one or more key fields within the data sample structure.
+These key fields serve as unique identifiers for each data sample published within the topic.
+to organize and manage the data samples, facilitating efficient access, retrieval, and filtering
+of data based on the specified key criteria.
+
+The ``@key`` annotation designates a field as a key for the given topic type, which can have zero or more key fields
+and can be applied to structure fields of various types:
+
+* Any primitive, such as booleans, integers, characters, strings or sequences.
+* Other existing or defined messages. If those have defined an inner key member, then those fields will we used as part of the key.
+  Otherwise, the key will be the concatenation of all the fields.
+
+In order to specify multiple keys, separate ``@key`` annotations are used.
+The following example shows how to define a keyed message using the IDL format in ROS 2:
+
+.. code-block:: bash
+
+    # KeyedMsgName.idl
+    module package_name {
+      module msg {
+        struct KeyedMsgName {
+          @key long key;
+          string data;
+        };
+      };
+    };
+
+.. note::
+
+    Currently, the only supported message format that can be annotated with ``@key`` is *.idl*.
+    Neither ``.msg`` nor ``.srv`` files support annotations.
+
+In this tutorial, we will demonstrate the use of topic keys in Vulcanexus by creating a simple example in
+which we will simulate a system with two sensors (each one identified by a key), and a controller.
```

Comment:
Weird sentence. Also, I think that the purpose of the tutorial should be the very first thing to be stated

Reply:
Will be left for the new introductory section